src/Microsoft.VisualStudio.ProjectSystem.Managed.TestServices/AssertEx.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Xunit
+{
+    public static class AssertEx
+    {
+        public static void CollectionLength<T>(IEnumerable<T> collection, int length)
+        {
+            Action<T>[] lengthArray = new Action<T>[length];
+            for (int i = 0; i < length; i++)
+            {
+                lengthArray[i] = delegate { };
+            }
+            Assert.Collection(collection, lengthArray);
+        }
+
+        public static void CollectionLength(IEnumerable collection, int length)
```

Comment:
Why?

Reply:
We have collections we get from MSBuild that only implement `IEnumerable `.  This is just a convenience method because `Assert.Collection` only accepts  `IEnumerable<T>`

Reply:
We should really just put this file in a NuGet package and put it up on MyGet (Roslyn already does this with a few other core files).